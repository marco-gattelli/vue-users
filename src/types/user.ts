import { Company } from "@/types/company";

export type User = {
  id: string;
  name: string;
  username: string;
  email: string;
  phone: string;
  website: string;
  company: Company;
};
