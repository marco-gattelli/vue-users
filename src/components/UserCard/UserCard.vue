<template>
  <li
    class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full"
    data-testid="user-card"
  >
    <div class="flex justify-between">
      <div class="flex flex-col gap-2">
        <span>Name: {{ user.name }}</span>
        <span>Email: {{ user.email }}</span>
        <span>Phone: {{ user.phone }}</span>
      </div>
      <div class="flex flex-col items-end gap-2">
        <span>Company: {{ user.company.name }}</span>
        <span class="italic">{{ user.company.catchPhrase }}</span>
      </div>
    </div>
  </li>
</template>

<script setup lang="ts">
import { User } from "@/types/user";

type Props = {
  user: User;
};
const { user } = defineProps<Props>();
</script>
