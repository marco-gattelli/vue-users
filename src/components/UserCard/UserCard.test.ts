import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import UserCard from "@/components/UserCard/UserCard.vue";
import { User } from "@/types/user";

describe("UserCard", () => {
  it("should render without crash", function () {
    // Arrange.
    const user: User = {
      email: "foo",
      username: "bar",
      id: "baz",
      name: "foobar",
      phone: "foobaz",
      website: "foofoo",
      company: {
        bs: "bar",
        name: "baz",
        catchPhrase: "foo",
      },
    };
    // Act.
    const { element } = mount(UserCard, { props: { user } });

    // Assert.
    expect(element).toMatchSnapshot();
  });
});
