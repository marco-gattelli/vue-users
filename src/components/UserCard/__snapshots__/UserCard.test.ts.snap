// Vitest Snapshot v1

exports[`UserCard > should render without crash 1`] = `
<li
  class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full"
  data-testid="user-card"
>
  <div
    class="flex justify-between"
  >
    <div
      class="flex flex-col gap-2"
    >
      <span>
        Name: foobar
      </span>
      <span>
        Email: foo
      </span>
      <span>
        Phone: foobaz
      </span>
    </div>
    <div
      class="flex flex-col items-end gap-2"
    >
      <span>
        Company: baz
      </span>
      <span
        class="italic"
      >
        foo
      </span>
    </div>
  </div>
</li>
`;
